app/src/main/java/org/groundplatform/android/ui/tos/TermsOfServiceFragment.kt

```diff
@@ -139,9 +140,19 @@ class TermsOfServiceFragment : AbstractFragment() {
     super.onViewCreated(view, savedInstanceState)
     lifecycleScope.launch {
       viewModel.navigateToSurveySelector.collect {
-        findNavController()
-          .navigate(SurveySelectorFragmentDirections.showSurveySelectorScreen(true))
+        activity?.intent?.data?.let { uri ->
+          if (uri.pathSegments.firstOrNull() == SURVEY_PATH_SEGMENT) {
+            val surveyId = uri.lastPathSegment
+            openSurveySelector(surveyId)
+          }
+        } ?: run { openSurveySelector(null) }
```

Comment:
We don't need to provide null here since it's the default value.

Reply:
fixed